resources/compass/templates/agent-configuration-job.yaml

```diff
@@ -0,0 +1,122 @@
+subjects:
+  - kind: ServiceAccount
+    name: compass-agent-configuration
+    namespace: {{ .Release.Namespace }}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: compass-agent-configuration
+  namespace: {{ .Release.Namespace }}
+rules:
+  - apiGroups: [""]
+    resources: [configmaps]
+    verbs: [create, get]
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: compass-agent-configuration
+  namespace: {{ .Release.Namespace }}
+  annotations:
+    "helm.sh/hook": post-install,post-upgrade
+    "helm.sh/hook-weight": "1"
+    "helm.sh/hook-delete-policy": hook-succeeded
+spec:
+  backoffLimit: 0
+  template:
+    metadata:
+      name: compass-agent-configuration
+      annotations:
+        sidecar.istio.io/inject: "false"
+    spec:
+      {{ if .Values.global.isLocalEnv }}
+      hostNetwork: true
+      hostAliases:
+        - ip: {{ .Values.global.minikubeIP }}
+          hostnames:
+            - "{{ .Values.global.gateway.host }}.{{ .Values.global.ingress.domainName }}"
+      {{ end }}
+      shareProcessNamespace: true
+      serviceAccountName: compass-agent-configuration
+      restartPolicy: Never
+      containers:
+        - name: compass-agent-configuration
+          image: "linkyard/kubectl:1.14.2"
+          command:
+            - bash
+            - -c
+            - |
+              MAX_RETRIES=60
+              DIRECTOR_URL=https://{{ .Values.global.gateway.host }}.{{ .Values.global.ingress.domainName }}/director/graphql
+
+              function checkIfDirectorReady() {
+                  local cnt=0
+                  set +o errexit
+
+                  while :
+                  do
+                    RESPONSE_CODE=$(curl -k -s "${DIRECTOR_URL}" \
+                      --write-out "%{http_code}\n" --output /dev/null \
+                      -H 'Content-Type: application/json' \
+                      -H 'tenant: {{ .Values.global.defaultTenant }}' \
+                      --data-binary '{
+                          "query":"query queryRuntimes{\n  result: runtimes {\n    data {\n      id\n    }\n  }\n}"
+                      }')
+                    if [[ "$RESPONSE_CODE" == "200" ]]
+                      then
+                        echo "Director ready."
+                        break
+                      else
+                        ((cnt++))
+                        if (( cnt > $MAX_RETRIES )); then
+                          echo "Max retries has been reached (retries $MAX_RETRIES). Exit."
+                          exit 1
+                        fi
+
+                        echo "Director not ready! StatusCode: '${RESPONSE_CODE}' - waiting 5s..."
+                        sleep 5
+                      fi
+                  done
+
+                  set -o errexit
+              }
+
+              kubectl get configmaps compass-agent-configuration -n compass-system > /dev/null 2>&1
+              if [[ $? -eq 1 ]]
+              then
```

Comment:
Please invert the condition and don't do else:
```bash
   if [[ $? -eq 0 ]]; then
      echo "Compass agent already configured and paired."
       exit 0
   fi

   ...
```

The code will be much simpler